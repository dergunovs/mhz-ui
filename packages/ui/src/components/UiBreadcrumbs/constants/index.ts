export const LINKS = [
  { url: '/', title: 'Main' },
  { url: '/category', title: 'Categories' },
  { url: '/cpu', title: 'CPU' },
];

export const DEFAULT_COLOR = 'default';
export const DEFAULT_ARIA_LABEL = 'Breadcrumb';
