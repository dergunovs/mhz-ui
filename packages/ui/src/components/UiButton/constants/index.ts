export const DEFAULT_LAYOUT = 'primary';
export const DEFAULT_TYPE = 'button';
export const DEFAULT_ICON = undefined;

export const DEFAULT_SLOT = 'Button label';
export const DEFAULT_ARIA_LABEL = 'Button';
