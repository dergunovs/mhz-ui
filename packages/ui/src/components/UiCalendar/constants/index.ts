import { ICalendarEvent } from '../interface';

export const EVENTS: ICalendarEvent<object>[] = [
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 1, text: 'Text' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 36%, rgb(235, 0, 123) 69%, rgb(251, 185, 0) 100%)',
  },
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 2, text: 'Text 2' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 41%, rgb(235, 0, 123) 71%, rgb(0, 134, 255) 100%)',
  },
];
