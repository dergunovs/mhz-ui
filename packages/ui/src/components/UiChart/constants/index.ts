export const LABELS = [
  '07.07 - 13.07',
  '14.07 - 20.07',
  '21.07 - 27.07',
  '28.07 - 03.08',
  '04.08 - 10.08',
  '11.08 - 17.08',
  '18.08 - 24.08',
];

export const DATASETS = [
  {
    data: [28, 16, 12, 4, 22, 16, 14],
    label: 'Грудь',
    borderColor: '#00CED1',
    backgroundColor: '#00CED1',
  },
  {
    data: [4, 6, 0, 2, 2, 0, 6],
    label: 'Ноги',
    borderColor: '#000080',
    backgroundColor: '#000080',
  },
  {
    data: [22, 18, 8, 10, 25, 18, 16],
    label: 'Плечи',
    borderColor: '#C41E3A',
    backgroundColor: '#C41E3A',
  },
  {
    data: [0, 6, 0, 6, 5, 4, 2],
    label: 'Пресс',
    borderColor: '#32CD32',
    backgroundColor: '#32CD32',
  },
  {
    data: [32, 24, 12, 10, 33, 24, 20],
    label: 'Руки',
    borderColor: '#DA70D6',
    backgroundColor: '#DA70D6',
  },
  {
    data: [16, 18, 8, 12, 26, 22, 20],
    label: 'Спина',
    borderColor: '#FF8C00',
    backgroundColor: '#FF8C00',
  },
];

export const TYPE = 'Line';
export const TITLE = 'Chart title';
