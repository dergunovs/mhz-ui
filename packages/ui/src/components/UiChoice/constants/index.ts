export const options = [1, 2, 3, 4, 5];
export const modelValue = options[2];
export const title = 'Заголовок';
export const isTall = true;
export const isInput = true;
