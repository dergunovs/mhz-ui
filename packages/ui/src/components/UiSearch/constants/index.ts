export const DEBOUNCE_TIME = 300;

export const MODEL_VALUE = '';
export const LANG = 'ru';
export const IS_SUCCESS = true;

export const RESULTS = {
  products: [
    { _id: '1', title: 'AMD Ryzen 7700X' },
    { _id: '2', title: 'AMD Ryzen 7900X' },
  ],
  manufacturers: [{ _id: '1', title: 'AMD' }],
};

export const SEARCH_SCHEME = [
  { type: 'products', labels: ['title'], url: '/products' },
  { type: 'manufacturers', labels: ['title'], url: '/manufacturers' },
];
