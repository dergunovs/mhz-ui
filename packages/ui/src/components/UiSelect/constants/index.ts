export const MODEL_VALUE = { _id: '1', title: 'Один', title_en: 'One' };

export const IS_FILTER = false;

export const OPTIONS = ['1', '2', '3', '4', '5', '6', '7', '8', '11', '111', '12'];

export const OPTIONS_OBJECTS = [
  { _id: '1', title: 'Один', title_en: 'One' },
  { _id: '2', title: 'Два', title_en: 'Two' },
  { _id: '3', title: 'Три', title_en: 'Three' },
];
