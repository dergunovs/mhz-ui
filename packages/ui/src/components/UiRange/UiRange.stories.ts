import { Meta, StoryObj } from '@storybook/vue3';
import { html } from 'mhz-helpers';

import { MODEL_VALUE, MIN, MAX } from './constants';

import { UiRange } from '@/components';

const meta: Meta<typeof UiRange> = {
  component: UiRange,
  args: {
    modelValue: MODEL_VALUE,
    min: MIN,
    max: MAX,
  },
  argTypes: {},
};

export default meta;

export const Primary: StoryObj<typeof UiRange> = {
  render: (args, { updateArgs }) => ({
    components: { UiRange },
    setup: () => ({ args, updateArgs }),

    template: html` <UiRange v-bind="args" @update:modelValue="update" /> `,

    methods: {
      update(value: [number, number]) {
        updateArgs({ modelValue: [...value] });
      },
    },
  }),
};
