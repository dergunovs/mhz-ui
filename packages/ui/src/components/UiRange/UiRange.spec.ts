import { DefineComponent, nextTick } from 'vue';
import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import UiRange from './UiRange.vue';
import { MIN, MAX, MODEL_VALUE } from './constants';

import { wrapperFactory } from '@/test';

const range = dataTest('ui-range');

let wrapper: VueWrapper<InstanceType<typeof UiRange>>;

beforeEach(() => {
  wrapper = wrapperFactory(UiRange, { modelValue: MODEL_VALUE, min: MIN, max: MAX });
});

enableAutoUnmount(afterEach);

describe('UiRange', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(UiRange)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('passes props to range component', async () => {
    expect(wrapper.find(range).attributes('modelvalue')).toBe(MODEL_VALUE.join(','));
    expect(wrapper.find(range).attributes('min')).toBe(MIN.toString());
    expect(wrapper.find(range).attributes('max')).toBe(MAX.toString());
  });

  it('emits update:modelValue event when slider value changes', async () => {
    const newValue: [number, number] = [200, 700];

    wrapper.findComponent<DefineComponent>(range).vm.$emit('update:modelValue', newValue);

    await nextTick();

    expect(wrapper.emitted('update:modelValue')).toBeTruthy();
    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual([newValue]);
  });
});
