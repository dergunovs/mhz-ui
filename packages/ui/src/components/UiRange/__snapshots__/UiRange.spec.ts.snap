// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiRange > matches snapshot 1`] = `
"<div class="_range_d6c3d5">
  <slider-stub modelvalue="100,600" disabled="false" min="10" max="1000" step="1" orientation="horizontal" direction="ltr" tooltips="true" options="[object Object]" merge="-1" classes="[object Object]" showtooltip="always" tooltipposition="bottom" lazy="true" aria="[object Object]" data-test="ui-range"></slider-stub>
</div>"
`;
